graylog2-web-interface/src/views/components/searchbar/TimeRangeDisplay.jsx

```diff
@@ -0,0 +1,101 @@
+// @flow strict
+import * as React from 'react';
+import { useEffect, useRef, useState } from 'react';
+import styled, { css, type StyledComponent } from 'styled-components';
+import moment from 'moment';
+
+import { type ThemeInterface } from 'theme';
+import { type TimeRange } from 'views/logic/queries/Query';
+import StoreProvider from 'injection/StoreProvider';
+
+type Props = {|
+  timerange: ?TimeRange,
+|};
+
+const EMPTY_RANGE = '----/--/-- --:--:--.---';
+const EMPTY_OUTPUT = { from: EMPTY_RANGE, until: EMPTY_RANGE };
+
+const ToolsStore = StoreProvider.getStore('Tools');
+
+const TimeRangeWrapper: StyledComponent<{}, ThemeInterface, HTMLParagraphElement> = styled.p(({ theme }) => css`
+  width: 100%;
+  padding: 3px 9px;
+  margin: 0 12px;
+  display: flex;
+  justify-content: space-around;
+  background-color: ${theme.colors.variant.lightest.primary};
+  align-items: center;
+  border-radius: 4px;
+  
+  > span {
+    flex: 1;
+  }
+  
+  code {
+    color: ${theme.colors.variant.dark.primary};
+    background: transparent;
+    font-size: ${theme.fonts.size.body};
+  }
+`);
+
+const dateOutput = (timerange: TimeRange) => {
+  let from = EMPTY_RANGE;
+
+  switch (timerange.type) {
+    case 'relative':
+      from = !timerange.range ? 'All Time' : moment()
+        .subtract(timerange.range * 1000)
+        .fromNow();
+
+      return {
+        from,
+        until: 'Now',
+      };
+
+    case 'absolute':
+    case 'keyword':
+      return { from: timerange.from, until: timerange.to };
+    default:
```

Comment:
If I am not mistaken `default` should `throw` an error. If someone is adding a new `timerange` type he would be thankful if the code is leading him to the places where he needs to insert the new type. And the current `default` should never occur otherwise.

Reply:
Updated!